spec/integration/util/windows/user_spec.rb

```diff
@@ -117,6 +117,28 @@ def expect_logon_failure_error(&block)
       it "should return false given a nil username and an incorrect password" do
         expect(Puppet::Util::Windows::User.password_is?(nil, bad_password)).to be_falsey
       end
+
+      context "with a correct password" do
+        it "should return true even if account restrictions are in place " do
+          Puppet::Util::Windows::User.stubs(:logon_user).raises(Puppet::Util::Windows::Error.new('', 1327))
```

Comment:
Under the circumstances I think these are probably OK

We might want to add one acceptance test around a disabled account (but not the other restrictions)